beta/nncf/tensorflow/quantization/quantizers.py

```diff
@@ -264,16 +267,62 @@ def build(self, input_shape, input_type, name, layer):
             'signed_var': signed
         }
 
-    def quantize(self, inputs, weights):
-        return symmetric_quantize(
-            inputs,
-            weights['scale_var'],
-            weights['signed_var'],
-            num_bits=self.num_bits,
-            per_channel=self.per_channel,
-            narrow_range=self.narrow_range,
-            eps=self._eps
-        )
+    def quantize(self, inputs, weights, training):
+        def _half_range_quantize():
+            return symmetric_quantize(
+                inputs,
+                weights['scale_var'],
+                weights['signed_var'],
+                num_bits=self.num_bits - 1,
+                per_channel=self.per_channel,
+                narrow_range=self.narrow_range,
+                eps=self._eps
+            )
+
+        def _full_range_non_narrow_quantize():
+            return symmetric_quantize(
+                inputs,
+                255 / 256 * weights['scale_var'],
+                weights['signed_var'],
+                num_bits=self.num_bits,
+                per_channel=self.per_channel,
+                narrow_range=self.narrow_range,
+                eps=255 / 256 * self._eps
+            )
+
+        def _full_range_narrow_quantize():
+            return symmetric_quantize(
+                inputs,
+                127 / 63 * weights['scale_var'],
+                weights['signed_var'],
+                num_bits=self.num_bits,
+                per_channel=self.per_channel,
+                narrow_range=self.narrow_range,
+                eps=127 / 63 * self._eps
+            )
+
+        def _default_quantize():
+            return symmetric_quantize(
+                inputs,
+                weights['scale_var'],
+                weights['signed_var'],
+                num_bits=self.num_bits,
+                per_channel=self.per_channel,
+                narrow_range=self.narrow_range,
+                eps=self._eps
+            )
+
+        if not self._half_range:
+            return _default_quantize()
+
+        if self.narrow_range:
```

Comment:
It looks like a working version, but only for num_bits=8. You should generalize on all num_bits values.